0.preprocess-sites/2.process-cells.py

```diff
@@ -0,0 +1,168 @@
+
+The following script will process cell painting metadata information for each site.
+
+The metadata include:
+
+1. Cell Counts per Quality Measure
+2. Metadata Information per Cell
+
+All sites are processed independently and results are saved in site-specific folders
+(in <OUTPUT_BASEDIR>/<BATCH>/paint set in site_processing_config)
+
+1.process-spots must be run before running this script.
+"""
+
+import os
+import sys
+import pathlib
+import argparse
+import pandas as pd
+
+sys.path.append(os.path.join("..", "scripts"))
+from config_utils import process_config_file
+from cell_quality_utils import CellQuality
+from paint_utils import (
+    load_single_cell_compartment_csv,
+    merge_single_cell_compartments
+)
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--config_file",
+    help="configuration yaml file for preprocessing pipeline",
+    default="site_processing_config.yaml",
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+#Defines the sections of the config file
+core_args = config["core"]
+cell_args = config["process-cells"]
+prefilter_args = config["prefilter"]
+spot_args = config["process-spots"]
+
+#Defines the variables set in the config file
+batch = core_args["batch"]
+batch_dir = core_args["batch_dir"]
+project_dir = core_args["project_dir"]
+quality_func = core_args["categorize_cell_quality"]
+ignore_files = core_args["ignore_files"]
+id_cols = core_args["id_cols"]
+compartments = core_args["compartments"]
+parent_cols = core_args["parent_cols"]
+
+prefilter_dir = prefilter_args["output_basedir"]
+prefilter_file = prefilter_args["prefilter_file"]
+
+foci_dir = spot_args["output_spotdir"]
+
+sort_col = cell_args["sort_col"]
+output_basedir = cell_args["output_basedir"]
+merge_info = cell_args["merge_columns"]
+metadata_merge_left = cell_args["metadata_merge_columns"]["left"]
+metadata_merge_right = cell_args["metadata_merge_columns"]["right"]
+metadata_merge_cells_left = cell_args["metadata_merge_columns"]["cells_left"]
+
+cell_quality = CellQuality(quality_func)
+cell_category_dict = cell_quality.define_cell_quality()
+cell_category_df = pd.DataFrame(cell_category_dict, index=["Cell_Class"])
+
+# Enables feature filtering by loading the Cell Painting feature file.
+# 0.prefilter-features.py must be run first
+try:
+    all_feature_df = pd.read_csv(prefilter_file, sep="\t")
+except FileNotFoundError:
+    raise FileNotFoundError(
+        "Error",
+        f"{prefilter_file} not found.  ",
+        "Perform 0.prefilter-features.py prefilter before continuing..."
+    )
+
+sites = [x for x in os.listdir(batch_dir) if x not in ignore_files]
+
+for site in sites:
+    try:
+        print(f"Now processing {site}...")
+        output_folder = pathlib.Path(output_basedir, batch, "paint", site)
+        os.makedirs(output_folder, exist_ok=True)
+
+        meta_output_file = pathlib.Path(output_folder, f"metadata_{site}.tsv.gz")
```

Comment:
Notes:
If we make a new PR that ever touches these files (they are not currently touched by anything downstream) we can update the meta_output_file in the PR so that nothing can break.

Reply:
These are actually both loaded by 7.visualize-cell-summary so we will need to move these to config when we re-write that script.

Reply:
great - it is ok to keep in this PR, we just need to remember to update later.